src/ncp/ncp_base.cpp

```diff
@@ -1859,16 +1863,114 @@ otError NcpBase::GetPropertyHandler_INTERFACE_COUNT(void)
     return mEncoder.WriteUint8(1); // Only one interface for now
 }
 
+#if OPENTHREAD_CONFIG_NCP_ENABLE_POWER_STATE_CONTROL
+
 otError NcpBase::GetPropertyHandler_POWER_STATE(void)
 {
-    return mEncoder.WriteUint8(SPINEL_POWER_STATE_ONLINE);   // Always online at the moment
+    otPlatPowerState state = otPlatGetPowerState(mInstance);
+    spinel_power_state_t spinelPowerState = SPINEL_POWER_STATE_ONLINE;
+
+    switch (state)
+    {
+    case OT_PLAT_POWER_STATE_OFFLINE:
+        spinelPowerState = SPINEL_POWER_STATE_OFFLINE;
+        break;
+
+    case OT_PLAT_POWER_STATE_DEEP_SLEEP:
+        spinelPowerState = SPINEL_POWER_STATE_DEEP_SLEEP;
+        break;
```

Comment:
Is it worth mentioning that the code should never return SPINEL_POWER_STATE_DEEP_SLEEP or SPINEL_POWER_STATE_OFFLINE since if the device were in either of these power states it would not be able to parse and reply to the command?

Reply:
True. Host should never be able to issue a get and get one of these as response.

This path in the code is used when when host issues `VALUE_SET(SPINEL_POWER_STATE)` to `DEEP_SLEEP`, the `NcpBase` prepares the spinel response using the `GetHandler` which at that time should return `DEEP_SLEEP` as the value.